beacon-chain/p2p/peers/status.go

```diff
@@ -427,8 +427,11 @@ func (p *Status) Decay() {
 // Ideally, all peers would be reporting the same finalized epoch but some may be behind due to their own latency, or because of
 // their finalized epoch at the time we queried them.
 // Returns the best finalized root, epoch number, and list of peers that are at or beyond that epoch.
-func (p *Status) BestFinalized(maxPeers int, ourFinalizedEpoch uint64) ([]byte, uint64, []peer.ID) {
+func (p *Status) BestFinalized(maxPeers int, ourFinalizedEpoch uint64) ([]byte, uint64, []peer.ID, error) {
 	connected := p.Connected()
+	if len(connected) == 0 {
+		return nil, 0, nil, errors.New("no connected peers")
+	}
```

Comment:
I am not sure that returning error is the way to go. It implies an invariant that `BestFinalized()` returns non-empty list of peers. I am not sure why this invariant is assumed, and believe that code using that function must check the len of returned peers (to assert that min. required peers are available).

Probably `BestFinalized()` method needs refactoring, but for me if method returns a list of peers and some extra data (like `targetRoot`) then reviewing extra data w/o first checking returned `len(peers)` is definitely something of a mis-use on the part of client (and I suppose, I check the len everywhere I've used the method).